scrapy/linkextractors/__init__.py

```diff
@@ -101,7 +101,7 @@ def _process_links(self, links):
         links = [x for x in links if self._link_allowed(x)]
         if self.canonicalize:
             for link in links:
-                link.url = canonicalize_url(urlparse(link.url))
+                link.url = canonicalize_url(link.url)
```

Comment:
I'm not 100% sure but I believe it was to have the same behavior as for [`SgmlLinkExtractor`](https://github.com/scrapy/scrapy/blob/155ea08ea1c50f347c21cb4635b36dd47141f4cc/scrapy/contrib/linkextractors/sgml.py#L150) prior to refactoring with `FilteringLinkExtractor`.